src/PowerShellEditorServices/Services/PowerShell/Execution/SynchronousPowerShellTask.cs

```diff
@@ -108,6 +144,17 @@ private IReadOnlyList<TResult> ExecuteNormally(CancellationToken cancellationTok
             // Other errors are bubbled up to the caller
             catch (RuntimeException e)
             {
+                if (e is PSRemotingTransportException)
+                {
+                    _ = System.Threading.Tasks.Task.Run(
+                        () => _psesHost.UnwindCallStack(),
+                        CancellationToken.None)
+                        .HandleErrorsAsync(_logger);
+
+                    _psesHost.WaitForExternalDebuggerStops();
+                    throw new OperationCanceledException("The operation was canceled.", e);
```

Comment:
Maybe explain this a bit? I guess I don't really know what `UnwindCallStack` does.

Reply:
It triggers a cancellation of every action we're currently doing. Basically if we get a transport error we want to try and abort out to essentially the "starting" state of a clean non-nested repl on our local runspace.

Reply:
Oh oh, I see. Nifty!